src/org/thoughtcrime/securesms/RegistrationActivity.java

```diff
@@ -175,6 +176,17 @@ public void onTextChanged(CharSequence s, int start, int before, int count) { }
 
             int certCheckFlags = DcHelper.getInt(this, "imap_certificate_checks");
             certCheck.setSelection(certCheckFlags);
+        } else if (getIntent() != null && getIntent().getBundleExtra(ACCOUNT_DATA) != null) {
+          // Companion app might have sent account data
+          Bundle b = getIntent().getBundleExtra(ACCOUNT_DATA);
+          String emailAddress = b.getString(CONFIG_ADDRESS);
+          String password = b.getString(CONFIG_MAIL_PASSWORD);
+          if (!TextUtils.isEmpty(emailAddress) && !TextUtils.isEmpty(password)) {
+            emailInput.setText(emailAddress);
+            passwordInput.setText(password);
+            updateProviderInfo();
+            onLogin();
+          }
```

Comment:
if email address and password are not set, we should show this as a toast or so to the user.

no need to have it translatable, it should happen to programmers using the Intent only, however, some hint would be great esp. then :)